goldens/public-api/angular/ssr/index.api.md

```diff
@@ -8,8 +8,9 @@ import { EnvironmentProviders } from '@angular/core';
 
 // @public
 export class AngularAppEngine {
-    getPrerenderHeaders(request: Request): ReadonlyMap<string, string>;
+    process(request: Request, requestContext?: unknown): Promise<Response | null>;
     render(request: Request, requestContext?: unknown): Promise<Response | null>;
+    serve(request: Request): Promise<Response | null>;
```

Comment:
Question: Is there still value in having separate methods for these operations? Should we just have a single `serve` method which does everything?

I'm a bit worried this API might be too complicated to use effectively and users might forget to call one of these steps or do them out of order. If we do need multiple methods here, we should at least document how we expect them to be used together.

Reply:
Here’s a revised version of your text:

---

- **`render`**: handles server-side rendering (SSR) requests.  
- **`serve`**: handles static site generation (SSG) requests.  
- **`process`**: manages both SSR and SSG requests.  

I’m considering renaming these methods or potentially consolidating them into a single method that addresses both SSG and SSR. My intention was to provide users with more control over request handling, but having a single method might simplify the process.

Reply:
My suggestion is to narrow this down to just `process` unless we have a need to expose the other two pieces. I feel like they just provide an opportunity for a caller to misuse them. We can always re-add them in the future if we find a need.

The one use case I can think of is that some apps might want to disable SSG serving because they expect a CDN layer in front on this server to do that and want a hard-error from Angular if that was misconfigured. That's a pretty specific use case though and might make more sense as an option to `process` than exposing a separate `render` method?

Reply:
Yes, that could be a fairly specific use case. As for naming, do you have any preference? `.render` feels a bit misleading, especially since it also deals with returning the pre-rendered page.

I'm leaning towards `.process`, `.handle`, or `.handleRequest`." any preference?